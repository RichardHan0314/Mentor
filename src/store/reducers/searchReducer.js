/* eslint-disable import/prefer-default-export */

import { SET_SEARCH_VALUE } from '../actions/searchActions';

const initialState = {
    searchValue: '',
    searchBy: 'title'
};

export function searchReducer(state = initialState, action) {
    switch (action.type) {
    case SET_SEARCH_VALUE:
        return {...state, searchValue: action.payload };
    default:
        return state;
    }
}
