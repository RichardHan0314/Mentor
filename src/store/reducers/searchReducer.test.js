import { searchReducer } from './searchReducer';
import {
    SET_SEARCH_VALUE
} from '../actions/searchActions';

const initialState = {
    searchValue: '',
    searchBy: 'title'
};

describe('search_reducer', () => {
    describe('INITIAL_STATE', () => {
        test('is correct', () => {
            expect(searchReducer(undefined, {})).toEqual(initialState);
        });
    });
      
    describe('SET_SEARCH_VALUE', () => {
        test('returns the correct state', () => {
            const action = { type: SET_SEARCH_VALUE, payload: 'MockSearchValue' };

            expect(searchReducer(undefined, action)).toMatchSnapshot();
        });
    });
});
