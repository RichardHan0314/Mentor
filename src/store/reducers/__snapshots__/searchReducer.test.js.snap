// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search_reducer SET_SEARCH_VALUE returns the correct state 1`] = `
Object {
  "searchBy": "title",
  "searchValue": "MockSearchValue",
}
`;
